### ISSUE TITLE
Feature Proposal: `/block` Command to Prevent Repeated File Addition Prompts

### ISSUE THREAD

**Issue by GeoffMillerAZ** [paul-gauthier/aider#1461]
#### Problem:

When working with large codebases, especially in a multi-directory setup, `aider` often suggests adding files from various locations in the workspace that aren’t relevant to the task at hand. For example, when troubleshooting a `requirements.txt` issue in the root directory, `aider` might repeatedly prompt the user to add similarly named files from different subdirectories (e.g., `tests/localstack/example/requirements.txt`), which can be frustrating.

Example of the issue:

```text
Add pulumi_lib/tests/localstack/example/requirements.txt to the chat? (Y)es/(N)o
 [Yes]: n
```

This prompt appears multiple times, despite the user already specifying not to add it. Constantly declining these prompts interrupts the flow of work.

#### Proposed Solution:

Introduce a new in-chat command, `/block` or `/block-add`, which allows the user to block specific files or file patterns from being suggested for addition in the future. This block should:
- Work with file names and patterns (supporting regex).
- Only affect files that haven’t been added yet (won’t remove files that are already part of the session).
- Allow for easy unblocking through either the `/drop` command or a new `/drop-block` command.

#### Command Usage:

Either of the following could be a good way to implement the command usage or perhaps something different.

- **/block [file_path_or_pattern]**: Blocks files matching the specified name or pattern from being suggested for addition.
- **/drop-block [file_path_or_pattern]**: Unblocks files matching the specified pattern, allowing them to be suggested for addition again.
  
By incorporating regex support, this allows users to easily block entire sets of files that follow a particular pattern (e.g., all `tests/requirements.txt` files).

#### Problem:

In addition to unwanted file addition prompts, `aider` can also suggest commands that may not be relevant or useful for the current context. Repeatedly declining these suggestions can disrupt the user's workflow.

For example, if `aider` frequently suggests commands that you don’t need or want to run, it can cause unnecessary distractions, similar to the file addition issue.

#### Proposed Solution:

Extend the blocking functionality to command suggestions with a new in-chat command called `/block-cmd`. This command would allow users to block specific commands (or command patterns) from being suggested by `aider`. This could be particularly useful when `aider` frequently suggests commands that are not relevant to the current task.

#### Command Usage:

- **/block-cmd [command_pattern]**: Blocks commands matching the specified pattern (supports regex) from being suggested during the session.
- **/drop-cmd [command_pattern]**: Unblocks commands matching the specified pattern, allowing them to be suggested again.

#### Example use case:

Let's say you're working on an Infrastructure as Code repo. And you would prefer that Aider never suggest a command that would cause the changes to happen yet. Like a `pulumi up`. Obviously, in this case you should be taking other precautions as well but its nice to have a bit more help.

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

The latest version of aider will remember if you say not to add a file. It won't ask about that file again during the session.

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to add a comment here and I will re-open. Or feel free to file a new issue any time.

**Comment by GeoffMillerAZ** [comment]
Cool! I think that works!!
